### PR TITLE
SQLite shredding

### PR DESCRIPTION
**Pull request by fehrenbach**
Seems to pretty much just work.
It's not extensively tested, though.

Related: #486, #322

### CODE REVIEW

File: core/evalir.ml [hunk 1]
```diff
@@ -678,8 +678,8 @@ struct
            | None -> computation env cont e
            | Some (db, p) ->
              begin
-               if db#driver_name() <> "postgresql"
-                 then raise (Errors.Runtime_error "Only PostgreSQL database driver supports shredding");
+               if not (List.mem (db#driver_name()) ["postgresql"; "sqlite3"])
```
Comment (jamescheney) [inline core/evalir.ml]: I wonder if we should instead add a test to `db` objects that tells us whether the db supports shredding, and report "this database driver XYZ does not support shredding".

File: links-sqlite3.opam [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-  [ "jbuilder" "build" "-p" name "-j" jobs ]
+  [ "dune" "subst" ] {pinned}
+  [ "dune" "build" "-p" name "-j" jobs ]
 ]
 
 depends: [
-  "jbuilder" {build}
+  "ocaml" {>= "4.06.0"}
+  "dune" {build}
   "sqlite3"
```
Comment (jamescheney) [inline links-sqlite3.opam]: Would be nice to be able to check that the actual installed SQLite version is > 3.25.0, but not sure if OPAM gives us a way to say this.  If not, then row_number queries will just mysteriously fail.

Comment (fehrenbach) [reply links-sqlite3.opam]: Ideally this would tie in with the runtime database driver test above. We don't need 3.25.0 for building, just for shredding.

**Comment by fehrenbach** [comment]
I agree with all of @jamescheney's comments above, but I'm unlikely to do anything about it. I was just curious whether it worked.

File: links-sqlite3.opam (hunk 2 above)
Comment (dhil) [reply links-sqlite3.opam]: There is a virtual OPAM package `conf-sqlite3`, but it doesn't seem to track the version number of system package.

**Comment by dhil** [comment]
Superseded by #495.

**Pull request closed without merging**